src/modules/privacy-idea-administration/privacy-idea-administration.controller.ts

```diff
@@ -89,4 +92,57 @@ export class PrivacyIdeaAdministrationController {
         );
         return new TokenStateResponse(piToken);
     }
+
+    @Post('assign/hardwareToken')
+    @HttpCode(HttpStatus.OK)
+    @ApiCreatedResponse({
+        description: 'The hardware token was successfully assigned.',
+        type: AssignHardwareTokenResponse,
+    })
+    @ApiBadRequestResponse({ description: 'Not found.' })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to assign hardware token.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to reset token.' })
+    @ApiNotFoundResponse({ description: 'Insufficient permissions to assign hardware token.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while assigning a hardware token.' })
+    @Public()
+    public async assignHardwareToken(
+        @Body() params: AssignHardwareTokenBodyParams,
+    ): Promise<AssignHardwareTokenResponse | undefined> {
+        try {
+            const result: AssignTokenResponse = await this.privacyIdeaAdministrationService.assignHardwareToken(
```

Comment:
Like in the other PR, this endpoint ignores the logged in user.

If this is not a part of this ticket, just resolve this comment

Reply:
check implemented